waspc/cli/src/Wasp/Cli/Command/CreateNewProject.hs

```diff
@@ -18,27 +18,41 @@ import Wasp.Cli.Command (Command, CommandError (..))
 import qualified Wasp.Cli.Command.Common as Command.Common
 import qualified Wasp.Cli.Common as Common
 import qualified Wasp.Data
-import Wasp.Util (indent)
+import Wasp.Util (indent, kebabToCamelCase)
 import qualified Wasp.Util.Terminal as Term
 
-newtype ProjectName = ProjectName {_projectName :: String}
+data ProjectInfo = ProjectInfo
+  { _projectName :: String,
+    _appName :: String
+  }
 
-createNewProject :: String -> Command ()
-createNewProject name
-  | isValidWaspIdentifier name = createNewProject' (ProjectName name)
+-- Returns either the ProjectInfo type that contains both the Project name
+-- and the App name (which might be the same), or an error describing why the name is invalid
+parseProjectInfo :: String -> Either String ProjectInfo
+parseProjectInfo name
+  | isValidWaspIdentifier appIdentifier = Right (ProjectInfo name appIdentifier)
   | otherwise =
+      Left $
+        intercalate
+          "\n"
+          [ "The project's name must be a valid Wasp identifier:",
+            indent 2 "- It can start with a letter or an underscore.",
+            indent 2 "- It can contain only letters, numbers, dash, or underscores.",
+            indent 2 "- It can't be a Wasp keyword."
+          ]
```

Comment:
It can actually be valid Wasp identifier + dashes.
Maybe we can skip this part where we mention it needs to be Wasp identifier and just list the conditions below, like this:
```
The project's name is not of a valid format!
- It can start with a letter or an underscore.
- It can contain only letters, numbers, dash, or underscores.
- It can't be a Wasp keyword.
```